Azure-Samples/aks-store-demo
# chore: securing terraform configs and bumping deps

## Purpose

* Securing Terraform configs and replacing tfsec with trivy
* Bumping deps within all projects
* Bumping base container images in Dockerfiles

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
*  Run trivy scan on the infra/terraform directory to ensure there are no config errors
*  Run the azd command to ensure deployment still works

```
git clone https://github.com/pauldotyu/aks-store-demo.git
cd aks-store-demo

# init new environment and set variables 
azd env new
azd env set AZURE_LOCATION uksouth
azd env set AKS_NODE_POOL_VM_SIZE standard_d2s_v5
azd env set DEPLOY_AZURE_WORKLOAD_IDENTITY true
azd env set DEPLOY_AZURE_OPENAI true
azd env set AZURE_OPENAI_LOCATION uksouth
azd env set DEPLOY_AZURE_SERVICE_BUS true
azd env set DEPLOY_AZURE_COSMOSDB true
azd env set AZURE_COSMOSDB_FAILOVER_LOCATION southeastasia
azd env set DEPLOY_OBSERVABILITY_TOOLS true

# provision azure infra
azd provision

# overwrite the registry to point to newer containers
azd env set AZURE_REGISTRY_URI ghcr.io/pauldotyu

# deploy the app
azd deploy
```

## What to Check
Verify that the following are valid
* Ensure the AKS Store Demo app is deployed, and you can access it via web browser

## Other Information
